framework/packages/abstract-core/src/base.rs

```diff
@@ -11,6 +12,9 @@ pub enum ExecuteMsg<BaseMsg, CustomExecMsg, ReceiveMsg = Empty> {
     /// An app request defined by a base consumer.
     Module(CustomExecMsg),
     /// IbcReceive to process IBC callbacks
+    /// In order to trust this, they should verify this comes from the a note contract.
```

Comment:
Oh so the callbacks come from the note and not the ibc client?

You can implement that check here: https://github.com/AbstractSDK/abstract/blob/aaf44ec8e0a284479908f19c89e5a18638759b39/framework/packages/abstract-app/src/endpoints/ibc_callback.rs

Reply:
Yes, that is a mistake in the comment, the callbacks come from the ibc client (because the ibc-client registers them to itself and redispatches them)